### PR TITLE
Note --disableProgress's argument

### PR DESCRIPTION
**Pull request by adamnovak**
@juklucas noted that we don't say we need an argument here, but we do need one.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * `--disableProgress` option argument is now documented.

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: docs/running/cliOptions.rst [hunk 1]
```diff
@@ -505,7 +505,8 @@ systems have issues!).
                         to provision mesos for autoscaling.
   --statusWait INT      Seconds to wait between reports of running jobs.
                         (default=3600)
-  --disableProgress     Disables the progress bar shown when standard error is
+  --disableProgress BOOL
+			Disables the progress bar shown when standard error is
                         a terminal.
```
Comment (mr-c) [inline docs/running/cliOptions.rst:510]: Wouldn't it be better to not require the bool at all?

Change `type=convert_bool` to `action="store_true"` and drop `metavar="BOOL"` at

https://github.com/DataBiosphere/toil/blob/67a8b471dd77db33c25bf9db071def18c8d53f1f/src/toil/options/common.py#L695-L697

(matching [what is done for `--stats`](https://github.com/DataBiosphere/toil/blob/67a8b471dd77db33c25bf9db071def18c8d53f1f/src/toil/options/common.py#L308C71-L309))

**Review by mr-c — CHANGES_REQUESTED** [review]

File: docs/running/cliOptions.rst (hunk 1 above)
Comment (adamnovak) [reply docs/running/cliOptions.rst:510]: That's basically what we used to do? @stxue1 you had a reason for switching this to take an argument, right?

If we're already making incompatible changes we might be better off with `--progress=True`/`--progress=False` and keeping the old `--disableProgress` without argument as a thing that doesn't show up in the config.

Comment (stxue1) [reply docs/running/cliOptions.rst:510]: When I was adding metavars, the previous implementation looked like it took a true/false argument. Though it doesn't make much sense to take a bool with its argument name, so it should probably be `action="store_true"` for this one.
https://github.com/DataBiosphere/toil/blob/e80db2cd9d8f800befd94bb454581fd781130ba3/src/toil/common.py#L1537-L1543

**Comment by adamnovak** [comment]
I'm closing this in favor of #4778.

**Pull request closed without merging**